papas/datatypes/Cluster.h

```diff
@@ -35,15 +35,26 @@ class Cluster {
                 When creating a merged cluster it should ideally be set to the total eneergy of the cluster
       */
   Cluster(const Cluster& cluster, unsigned int index, IdCoder::ItemType type, char subtype = 'u', float val = 0.0);
+
+  /** Constructor: makes new merged cluster
+   @param[in]  overlappingClusters list of clusters to be merged, they must have same type and subtype and must be
+   simple
+               unmerged clusters.
+   @param[in]  index of the collection into which the cluster is to be stored
+   @param[in]  subtype subtype of cluster to be created eg 'm' for merged,
+   */
+  Cluster(std::list<const Cluster*> overlappingClusters, unsigned int index, char subtype = 'm');
```

Comment:
OverlappingClusters list is made in MergeClusters.cpp:
https://github.com/alicerobson/papas/blob/3ee37b9231495769be2a8aa0083a3cc2b5cbe62d/papas/reconstruction/src/MergeClusters.cpp#L46
I decided to have a go at implementing it using shared_ptr. See my sharedclust branch https://github.com/alicerobson/papas/tree/sharedclust
It runs but produces a segmentation fault ....